File: 12.py
class Solution:
    def intToRoman(self, num: int) -> str:
        if num == 0: return ''
        hashmap = {
            1 : 'I',
            2 : 'II',
            3 : 'III',
            4 : 'IV',
            5 : 'V',
            9 : 'IX',
            10 : 'X',
            40 : 'XL',
            50 : 'L',
            90 : 'XC',
            100 : 'C',
            400 : 'CD',
            500 : 'D',
            900 : 'CM',
            1000 : 'M'
        }
        nums = sorted(hashmap.keys(), reverse=True)
        # print(nums)
        ret = ''
        while num != 0:
            for temp in nums:
                if num - temp >= 0:
                    break
            num -= temp
            ret += hashmap[temp]
        return ret

def main():
    x = 1994
    print(Solution().intToRoman(x))

if __name__ == "__main__":
    main()

File: leetcode-cpp/12.cpp
#include <iostream>
#include <map>
#include <string>
using namespace std;

class Solution {
public:
    string intToRoman(int num) {
        int nums[] = {1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1};
        string strs[] = {"M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"};
        int i = 0;
        string ret = "";
        while (num != 0){
            while (num - nums[i] < 0)
                i++;
            num -= nums[i];
            ret.append(strs[i]);
        }
        return ret;
    }
};

int main(){
    Solution s;
    int a = 1994;
    cout << s.intToRoman(a) <<endl;
    return 0;
}
